### PR TITLE
[New Rule] New or Modified Federation Domain

### LINKED ISSUE #1162: [New Rule] New or Modified Federation Domain
## Description

Federation domains can be used to create a trust between O365 and an external identity provider.

### Target indexes

`filebeat-*", "logs-o365*`

### Platforms

Microsoft 365

### Query

```
event.dataset:o365.audit and 
event.provider:Exchange and event.category:web and event.action:("Set-AcceptedDomain" or 
"Set-MsolDomainFederationSettings" or "Add-FederatedDomain" or "New-AcceptedDomain" or "Remove-AcceptedDomain" or 
"Remove-FederatedDomain") and event.outcome:success
```

#### ATTACK TACTIC
Persistence

#### ATTACK TECHNIQUE
Account Manipulation

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1162

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/microsoft-365/persistence_new_or_modified_federation_domain.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+author = ["Elastic", "Austin Songer"]
+description = """
+Identifies a new or modified federation domains can be used to create a trust between O365 and an external identity
+provider.
+"""
+index = ["filebeat-*", "logs-o365*"]
+language = "kuery"
+name = "New or Modified Federation Domain"
+risk_score = 21
+rule_id = "684554fc-0777-47ce-8c9b-3d01f198d7f8"
```
Comment (brokensound77) [inline rules/microsoft-365/persistence_new_or_modified_federation_domain.toml:16]: Please provide references if possible

File: rules/microsoft-365/persistence_new_or_modified_federation_domain.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[metadata]
+creation_date = "2021/05/17"
+maturity = "development"
+updated_date = "2021/05/17"
+
+[rule]
+author = ["Elastic", "Austin Songer"]
```
[the remark this answers is not in the document]
Comment (austinsonger) [reply rules/microsoft-365/persistence_new_or_modified_federation_domain.toml:7]: Oh okay. Good to know.

File: rules/integrations/o365/persistence_new_or_modified_federation_domain.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+[[rule.threat.technique]]
+reference = "https://attack.mitre.org/techniques/T1098/"
+name = "Account Manipulation"
+id = "T1098"
+
+
+[rule.threat.tactic]
+reference = "https://attack.mitre.org/tactics/TA0003/"
+name = "Persistence"
+id = "TA0003"
```
Comment (w0rk3r) [inline rules/integrations/o365/persistence_new_or_modified_federation_domain.toml:52]: This one fits better:
https://attack.mitre.org/techniques/T1484 and https://attack.mitre.org/techniques/T1484/002/

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Hey @austinsonger, left a requested change, after that, LGTM

**Review by w0rk3r — APPROVED** [review]

**Pull request merged**